lib/srv/discovery/discovery.go

```diff
@@ -756,13 +768,79 @@ func (s *Server) handleEC2Instances(instances *server.EC2Instances) error {
 	// instances.Rotation is true whenever the instances received need
 	// to be rotated, we don't want to filter out existing OpenSSH nodes as
 	// they all need to have the command run on them
-	if !instances.Rotation {
+	//
+	// Integration/EICE Nodes don't have heartbeat.
+	// Those Nodes must not be filtered, so that we can extend their expiration and sync labels.
+	if !instances.Rotation && instances.Integration == "" {
 		s.filterExistingEC2Nodes(instances)
 	}
 	if len(instances.Instances) == 0 {
 		return trace.NotFound("all fetched nodes already enrolled")
 	}
 
+	switch {
+	case instances.Integration != "":
+		s.handleEC2UsingEICE(instances)
+	default:
+		if err := s.handleEC2RemoteInstallation(instances); err != nil {
+			return trace.Wrap(err)
+		}
+	}
+
+	if err := s.emitUsageEvents(instances.MakeEvents()); err != nil {
+		s.Log.WithError(err).Debug("Error emitting usage event.")
+	}
+
+	return nil
+}
+
+// handleEC2UsingEICE creates a Teleport Node (EICE subkind) from the instances list.
+func (s *Server) handleEC2UsingEICE(instances *server.EC2Instances) {
```

Comment:
Nit: I'd probably name this "heartbeatEICInstance" or something like this because that's what it effectively does.